src/screens/WelcomeScreen.js

```diff
@@ -320,57 +331,75 @@ function colorAnimation(rValue, fromShadow) {
 }
 
 export default function WelcomeScreen() {
+  const { replace, navigate } = useNavigation();
   const contentAnimation = useAnimatedValue(1);
-  const importButtonAnimation = useAnimatedValue(1);
   const hideSplashScreen = useHideSplashScreen();
+  const createWalletButtonAnimation = useAnimatedValue(1);
+  const [userData, setUserData] = useState(null);
 
   useEffect(() => {
-    hideSplashScreen();
-    Animated.parallel([
-      ...traversedRainbows.map(({ value, delay = 0 }) =>
-        Animated.spring(value, { ...springConfig, delay })
-      ),
-      Animated.sequence([
-        Animated.timing(contentAnimation.current, {
-          duration: 120,
-          easing: Easing.bezier(0.165, 0.84, 0.44, 1),
-          toValue: 1.2,
-        }),
-        Animated.spring(contentAnimation.current, {
-          friction: 8,
-          tension: 100,
-          toValue: 1,
-        }),
-      ]),
-      Animated.loop(
-        Animated.sequence([
-          Animated.timing(importButtonAnimation.current, {
-            duration: 1000,
-            toValue: 1.02,
-            useNativeDriver: true,
-          }),
-          Animated.timing(importButtonAnimation.current, {
-            duration: 1000,
-            toValue: 0.98,
-            useNativeDriver: true,
-          }),
-        ])
-      ),
-    ]).start();
+    const initialize = async () => {
+      try {
+        logger.log('downloading iCloud backup info...');
+        const isSimulator = await DeviceInfo.isEmulator();
+        if (!isSimulator) {
+          const data = await fetchUserDataFromCloud();
+          setUserData(data);
+          logger.log('Downloaded iCloud backup info');
+        }
+      } catch (e) {
+        logger.log('error getting userData', e);
```

Comment:
Not sure if this is handled later, but wondering if we should have retry around fetching the user data from cloud downstream in the code (or show a failure message when we can't get it but the user expects to be able to restore)

Reply:
That's fair. @christianbaroni  What do you think about this scenario? Launching the app and error while trying to check for iCloud backups. Currently we're hiding the option. Should we have a "not available. try again later" state?

Reply:
Just a note to myself for later: if the fetchUserDataFromCloud doesn't return as expected, then the RestoreSheetFirstStep will not show the "Restore from iCloud" option even tho they might have had the backup from icloud